networks/monza/monza-config/src/lib.rs

```diff
@@ -1,20 +1,36 @@
-#[derive(Debug, Clone, PartialEq, Eq)]
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct Config {
+	#[serde(flatten)]
```

Comment:
This is not future-proof: if something needs to be added besides the execution units' config, the file format will need to be broken.

Reply:
> Overall, I suggest we adopt (as this PR currently and implicitly does) a standard of config parameters similar to that used with protobuf message types. That is fields, or at least wrapping structs should be globally unique. This allows for flattening at higher-order configs and thus for configs to continue to be read from a single config.toml. m1-da-light-node parameters can be improved in this regard.

Does this not cover that? Essentially, we have to assume that every field that needs to be read from a file does not conflict in the global namespace. So, it's flatten at every level where reading from `config.toml` is valid.

Reply:
Understood. As long as we manage the fields to be unique, this is workable.